eng/scripts/Copy-ApiVersion.ps1

```diff
@@ -1,77 +1,175 @@
+Write-Host "Service Type: $ServiceType" -ForegroundColor Green
+Write-Host "Provider: $Provider" -ForegroundColor Green
+Write-Host "Base Version: $BaseVersion" -ForegroundColor Green
+Write-Host "New Version: $NewVersion" -ForegroundColor Green
+Write-Host "----------------------------------------" -ForegroundColor Green
+
+Write-Verbose "Copying $oldDirectory to $newDirectory"
 
 # Copy the specs and create and initial commit to make it easier to diff changes to the previous version in a PR.
 Copy-Item $oldDirectory $newDirectory -Recurse -Force
 
 git add $newDirectory
 @"
-Copy files from $FromVersion
+Copy files from $BaseVersion
 
 Copied the files in a separate commit.
 This allows reviewers to easily diff subsequent changes against the previous spec.
 "@ | git commit --file=-
 
-# Replace the $oldVersion with the $newVersion within all files in $newDirectory.
+# Replace the $oldApiVersion with the $newApiVersion within all files in $newDirectory.
 foreach ($file in Get-ChildItem $newDirectory -File -Recurse) {
     Write-Verbose "Replacing any API versions in $file"
     $content = $file | Get-Content -Raw
-    $content -replace $oldVersion, $newVersion | Set-Content $file.FullName
+    $content -replace $oldApiVersion, $newApiVersion | Set-Content $file.FullName
 }
 
 # Commit just the version changes.
 git add $newDirectory
 @"
-Update version to $ToVersion
+Update version to $NewVersion
 
-Updated the API version from $FromVersion to $ToVersion.
+Updated the API version from $BaseVersion to $NewVersion.
 "@ | git commit --file=-
 
-Write-Host ''
-Write-Host '!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!' -ForegroundColor 'Yellow'
-Write-Host '!!! IMPORTANT: Action Required !!!' -ForegroundColor 'Yellow'
-Write-Host '!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!' -ForegroundColor 'Yellow'
-Write-Host ''
-Write-Host "You must manually update the default version and copy $FromVersion sections from any markdown files in $readmeDirectory."
-Write-Host 'When complete, commit those changes as shown below, push to the upstream repo, and submit a pull request.'
-Write-Host ''
-Write-Host '    git commit -am"Updated readme files"'
-Write-Host ''
\ No newline at end of file
+# Update readme file in the service type directory
+$jsonFiles = Get-ChildItem $newDirectory -Filter '*.json' | Select-Object -ExpandProperty Name
+$readmeFile = Get-ChildItem $readmeDirectory -Filter 'readme.md'
+
+$readmeContentBlock = @"
+### Tag: package-$newApiVersion
+
+These settings apply only when ``--tag=package-$newApiVersion`` is specified on the command line.
+
+```````yaml `$(tag) == 'package-$newApiVersion'
+input-file:
+$(
+foreach ($jsonFile in $jsonFiles) {
+  "  - $Provider/$newApiVersionStatus/$newApiVersion/$jsonFile"
+}
+)
+```````
+
+"@
+
+if ($readmeFile) {
+    $readmeContent = $readmeFile | Get-Content -Raw
+    $readmeContent -replace '(?s)(### tag: package.*)', "$readmeContentBlock`n`$1" | Set-Content $readmeFile.FullName -NoNewline
```

Comment:
Isn't this regex going to replace the entire file as opposed to just adding one new version?

Reply:
It captures everything from the first tag until the end of the file. I did it this way so I can be sure it always captures the first one, and then it is replaced with the new version tag and the captured group appended.

I tried to do it without using `(?s)` but I couldn't capture only the first tag in several scenarios, specially in the ones where the basic configuration section is larger.